conmon-rs/server/src/rpc.rs

```diff
@@ -105,19 +115,30 @@ impl conmon::Server for Server {
 
         Promise::from_future(async move {
             match child_reaper
-                .create_child(&runtime, &args, console, pidfile)
+                .create_child(&runtime, &args, Some(&console), pidfile)
                 .await
             {
                 Ok(grandchild_pid) => {
                     let mut resp = results.get().init_response();
                     // register grandchild with server
                     let child = Child::new(id, grandchild_pid, child.exit_paths);
-                    let _ = child_reaper.watch_grandchild(child);
+                    capnp_err!(child_reaper.watch_grandchild(child))?;
+
                     // TODO return the grandchild exit code
-                    // TODO return the stdout and stderr of the child
+
+                    let mut stdio = vec![];
+                    for msg in console.message_rx() {
+                        match msg {
+                            Message::Data(s) => stdio.extend(s),
```

Comment:
are we at risk of DOSing the server if this exec request takes a long time?

Reply:
I took the time out into account, this should prevent such attacks.